content/developer/tutorials/server_framework_101/05_connect_the_dots.rst

```diff
@@ -0,0 +1,849 @@
+
+.. spoiler:: Solution
+
+   .. code-block:: python
+      :caption: `real_estate_property.py`
+      :emphasize-lines: 1
+
+      total_area = fields.Integer(string="Total Area", compute='_compute_total_area', store=True)
+
+   .. code-block:: text
+      :caption: terminal
+
+      $ psql -d tutorials
+
+      tutorials=> \d real_estate_property
+                                                  Table "public.real_estate_property"
+            Column       |            Type             | Collation | Nullable |                     Default
+      -------------------+-----------------------------+-----------+----------+--------------------------------------------------
+       [...]
+       total_area        | integer                     |           |          |
+
+.. _tutorials/server_framework_101/search_methods:
+
+Search computed fields
+----------------------
+
+As mentioned before, computed fields cannot be used in search queries unless they are stored in the
+database. This limitation arises because searches are performed at the database level, which is not
+aware of the existence of non-stored computed fields. However, storing every field that we wish to
+search on would be inefficient. **Search methods** provide a way to overcome this limitation.
+
+To enable searching on a computed field, a Python method must be defined and linked to the field
+declaration using the `search` argument. This method receives the search query's `operator` and
+`value` and should return a search domain that specifies how the query should filter records. The
+domain must be constructed using stored fields only.
+
+.. example::
+   In the example below, a search method is added to allow searching on the `is_profitable` field.
+
+   .. code-block:: python
+
+      is_profitable = fields.Boolean(
+          string="Profitable", compute='_compute_is_profitable', search='_search_is_profitable'
+      )
+
+      def _search_is_profitable(self, operator, value):
+          if (operator == '=' and value is True) or (operator == '!=' and value is False):
+              return [('margin', '>', 0)]
+          elif (operator == '=' and value is False) or (operator == '!=' and value is True):
+              return [('margin', '<=', 0)]
+          else:
+              raise NotImplementedError()
+
+   .. note::
+      - Search methods return a search domain that matches the computation of the searched field.
+      - It is not required to implemented all search operators.
+
+Our real estate app would be more powerful if we could add a set of search filters based on computed
+fields to the property views. Let’s leverage search methods to achieve this.
+
+.. exercise::
+   Add the following search filters to the real estate property views:
+
+   - **Stalled**: The property is past its availability date.
+   - **Priority**: The property has an offer that expires in less than two days. **Don't** use a
+     search method for this one.
```

Comment:
but in the solutions, there is a search method, why?

Reply:
I wanted the reader to store the field but it's better to write a search method in the search methods section. Instead, the reader is tasked to not write a search method in the exercise of the related fields now :)